.github/workflows/build.yml

```diff
@@ -38,28 +38,7 @@ jobs:
       run: RUSTFLAGS="-Dwarnings" cargo clippy --workspace
 
     - name: Test
-      run: cargo test --verbose --workspace
-
-    - name: Test with rust crypto libraries
-      run: cargo test --workspace --no-default-features --features w3c,tezos,eip,solana
-
-    - name: Test with openssl and ring
-      run: cargo test --workspace --no-default-features --features ring,openssl,w3c,tezos,eip,solana
-
-    - name: Test with secp256k1
-      run: |
-        cargo test --verbose --workspace --features secp256k1
-        cargo test --verbose --manifest-path did-key/Cargo.toml --features secp256k1,ssi/ring
-        cargo test --verbose --manifest-path did-tezos/Cargo.toml --features secp256k1,ssi/ring
-
-    - name: Test with p256
-      run: |
-        cargo test --verbose --workspace --features secp256r1
-        cargo test --verbose --manifest-path did-key/Cargo.toml --features secp256r1,ssi/secp256r1,ssi/ring
-        cargo test --verbose --manifest-path did-tezos/Cargo.toml --features secp256r1,ssi/secp256r1,ssi/ring
-
-    - name: Test DID Resolution HTTP(S) Binding
-      run: cargo test --manifest-path ssi-dids/Cargo.toml --features http
+      run: cargo hack test --each-feature --workspace --exclude-features "tor-tests" -- --test-threads=4
```

Comment:
do we want to do `--feature-powerset` or `--group-features` instead of `--each-feature`? I think this may exclude some combinations and tests

Reply:
Depends if you want to run tests for 17,421 combinations -- I checked and there didn't seem to be obvious combination of features. I will double check on tests though, making sure e.g. secp256k1 tests run for both the `secp256k1` and `ring` features

Reply:
I added another step in the CI to run the tests with all features enabled, that should cover any build/test problem